### DISCUSSION TITLE
Empty results from local (world data) service versus public server

### DISCUSSION THREAD

**Discussion by brad-4k** [osm-search/Nominatim#2875] [General]
Can I jump in on this discussion?

I'm running into a similar issue but... worse. I performed a whole world import as well and but there are some addresses where I get nothing in return (on my setup) but there is a decent answer from the "public" nominatim server.

It's not all addresses but a few. How might I verify that my import is correct?

Query on my server:
http://localhost:8080/search?q=38+Abel+Tasmanstraat,Rotterdam,NL+3165&format=geojson

Answer:
```
{
  "type": "FeatureCollection",
  "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  "features": [
    
  ]
}
```

Query on the public server:
https://nominatim.openstreetmap.org/search?q=38+Abel+Tasmanstraat,Rotterdam,NL+3165&format=geojson

Answer:
```
{
  "type": "FeatureCollection",
  "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  "features": [
    {
      "type": "Feature",
      "properties": {
        "place_id": 101239458,
        "osm_type": "way",
        "osm_id": 7518933,
        "display_name": "Abel Tasmanstraat, Distripark Eemhaven, Rotterdam-Albrandswaard, Albrandswaard, South Holland, Netherlands, 3165 AM, Netherlands",
        "place_rank": 26,
        "category": "highway",
        "type": "unclassified",
        "importance": 0.5
      },
      "bbox": [
        4.4134436,
        51.8678575,
        4.4243001,
        51.8703737
      ],
      "geometry": {
        "type": "Point",
        "coordinates": [
          4.4161459,
          51.8688234
        ]
      }
    }
  ]
}
```

**Comment by lonvia** [comment]
Which version of Nominatim do you use? You need the improved postcode parsing support from 4.1.0 in order for the query to work.

**Reply by brad-4k** [reply]
Looks like 4.0. It is many months old.  Let me update everything and test again.
Will I have to regenerate the data?

**Reply by brad-4k** [reply]
I may have answered my own question :)
https://nominatim.org/release-docs/latest/admin/Migration/

**Reply by lonvia** [reply]
Indeed. Getting the new postcode handling requires a reimport, even if you are already using the ICU tokenizer.

Just as a bit of background: the addresses on that street have a postcode of '3165AM' mapped. Your version of Nominatim would just add the postcode as is to the index and then it won't match against '3165'. The new version knows about postcode formats in the Netherlands and does some clever splitting and normalisation so that it now matches against various spelling variants of the postcode.

**Reply by brad-4k** [reply]
This may have to be another topic all-together, and I'm sure I found this someplace but can't now, but I'm using the docker image (mediagis/nominatim:4.1) and I'm trying to import planet-latest.osm.pbf.  I've done this before but...  those braincells have died.

How does one properly size the docker storage to fit the postgres storage for planet-latest.osm.pbf?

**Reply by brad-4k** [reply]
... and again... I think I answered my own question:

https://github.com/mediagis/nominatim-docker/tree/master/4.1#persistent-container-data

**Reply by brad-4k** [reply]
Thank you Sarah (@lonvia ).  I let the update run over the weekend and ran some initial tests and I'm getting better results.  Thank you and the team for all the hard work.
